BEIMA.Backend/DeviceFunctions/AddDevice.cs

```diff
@@ -16,45 +17,80 @@ namespace BEIMA.Backend.DeviceFunctions
+        }
+
         /// <summary>
         /// Handles device create request.
         /// </summary>
         /// <param name="req">The http request.</param>
         /// <param name="log">The logger to log to.</param>
         /// <returns>An http response containing the id of the newly created device.</returns>
         [FunctionName("AddDevice")]
-        public static async Task<IActionResult> Run(
+        public async Task<IActionResult> Run(
             [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "device")] HttpRequest req,
             ILogger log)
         {
             log.LogInformation("C# HTTP trigger function processed a device post request.");
 
+            var reqForm = await req.ReadFormAsync();
+
             Device device;
             try
             {
-                string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
-                dynamic data = JsonConvert.DeserializeObject(requestBody);
-                device = new Device(ObjectId.GenerateNewId(),
-                                    ObjectId.Parse((string)data.deviceTypeId),
-                                    (string)data.deviceTag,
-                                    (string)data.manufacturer,
-                                    (string)data.modelNum,
-                                    (string)data.serialNum,
-                                    (int)data.yearManufactured,
-                                    (string)data.notes);
+                var data = JsonConvert.DeserializeObject<AddDeviceRequest>(reqForm["data"]);
+                device = new Device(
+                    ObjectId.GenerateNewId(),
+                    ObjectId.Parse(data.DeviceTypeId),
+                    data.DeviceTag,
+                    data.Manufacturer,
+                    data.ModelNum,
+                    data.SerialNum,
+                    data.YearManufactured,
+                    data.Notes
+                );
+
+                device.SetLocation(
+                    ObjectId.Parse(data.Location.BuildingId),
+                    data.Location.Notes,
+                    data.Location.Latitude,
+                    data.Location.Longitude
+                );
 
-                device.SetLocation(ObjectId.Parse((string)data.location.buildingId),
-                                   (string)data.location.notes,
-                                   (string)data.location.latitude,
-                                   (string)data.location.longitude);
-                // TODO: include device type attributes
+                device.Fields = data.Fields;                
             }
             catch (Exception)
             {
                 return new BadRequestObjectResult(Resources.CouldNotParseBody);
             }
+
+            // Store attached files and add file uid to device
+            foreach (var file in reqForm.Files)
+            {
+                var fileUid = await _storage.PutFile(file);
```

Comment:
Is there any verification that needs to be done on files before putting them in?

Reply:
Added making sure they have the right name and that the file doesn't have a size of zero